src/app/maps/m6/m6.component.ts

```diff
@@ -0,0 +1,455 @@
+            this.LAYER_NAME,
+            this.SOURCE_NAME
+          );
+
+          this.sample = null;
+          this.theMap.on('mouseenter', this.LAYER_NAME, (e) => {
+            this.theMap.getCanvas().style.cursor = 'pointer';
+            // Copy coordinates array.
+            const coordinates = e.features[0].geometry['coordinates'].slice();
+            while (Math.abs(e.lngLat.lng - coordinates[0]) > 180) {
+              coordinates[0] += e.lngLat.lng > coordinates[0] ? 360 : -360;
+            }
+            this.sample = e.features[0].properties;
+            this.Location = this.sample.Location;
+            this.Chapter = this.sample.Chapter;
+            popup
+              .setLngLat(coordinates)
+              .setHTML(this.Location + ' - ' + this.Chapter)
+              .addTo(this.theMap);
+          });
+          this.theMap.on('mouseleave', this.LAYER_NAME, () => {
+            this.theMap.getCanvas().style.cursor = '';
+            popup.remove();
+          });
+
+          var items;
+          items = this.sample.Images;
+          var group = 0;
+          var images = [];
+          for (let i = 0; i < items?.length; i++) {
+            if (i > 0 && i % 4 === 0) group++;
+            if (!Array.isArray(images[group])) images[group] = [];
+            images[group].push(items[i]);
+          }
+          this.images = images;
+          this.theMap.on('click', this.LAYER_NAME, (e) => {
+            this.sample = e.features[0].properties;
+            this.sample = {
+              ...this.sample,
+              Images: [
+                ...JSON.parse(this.sample.Images).map((image) =>
+                  image.width <= image.height
+                    ? { ...image, newClass: true }
+                    : { ...image, newClass: false }
+                ),
+              ],
+            };
+            this.accordion = this.sample.accordion;
+            this.name = this.sample.name;
+            this.pics = this.sample.Images;
+            this.description = this.sample.description;
+            this.Location = this.sample.Location;
+            this.Chapter = this.sample.Chapter;
+            this.Notes = this.sample.Notes;
+            this.Shortcuts =
+              this.sample.Shortcuts === undefined
+                ? this.sample.Shortcuts?.json()
+                : JSON.parse(this.sample.Shortcuts);
+            this.title = this.Location + ' - ' + this.Chapter;
+            this.features = samples.features;
+
+            this.info = false;
+            location.hash = '#' + this.sample.id;
+
+            var items = this.sample.Images;
+            var group = 0;
+            var images = [];
+            for (let i = 0; i < items.length; i++) {
+              if (i > 0 && i % 4 === 0) group++;
+              if (!Array.isArray(images[group])) images[group] = [];
+              images[group].push(items[i]);
+            }
+            this.images = images;
+            setTimeout(function () {

```

Comment:
very hacky and won't always work - must be a better way for this, no?

Reply:
it's good, I think we will use it like this for now